src/Managers/ProcessDRM.vala

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ */
+
+public class Monitor.ProcessDRM {
+    /** Time spent busy in nanoseconds by the render engine executing
+     * workloads from the last time it was read
+     */
+    private uint64 last_engine_render;
+    private uint64 last_engine_gfx;
+
+
+    public double gpu_percentage { get; private set; }
+
+    private int pid;
+    private int update_interval;
+
+    public ProcessDRM (int pid, int update_interval) {
+        this.pid = pid;
+        this.update_interval = update_interval;
+
+        last_engine_render = 0;
+        last_engine_gfx = 0;
+    }
+
+    public void update () {
+        string path_fdinfo = "/proc/%d/fdinfo".printf (pid);
+        string path_fd = "/proc/%d/fd".printf (pid);
+
+
+        var drm_files = new Gee.ArrayList<GLib.File ?> ();
+
+        try {
+            Dir dir = Dir.open (path_fdinfo, 0);
+            string ? name = null;
+
+            while ((name = dir.read_name ()) != null) {
+
+                // skip standard fds
+                if (name == "0" || name == "1" || name == "2") {
+                    continue;
+                }
+                string path = Path.build_filename (path_fdinfo, name);
+
+                int fd_dir_fd = Posix.open (path_fd, Posix.O_RDONLY | Posix.O_DIRECTORY, 0);
+                bool is_drm = is_drm_fd (fd_dir_fd, name);
+                Posix.close (fd_dir_fd);
+
+                if (is_drm) {
+                    var drm_file = File.new_for_path (path);
+                    drm_files.add (drm_file);
+                }
+            }
+        } catch (FileError err) {
+            if (err is FileError.ACCES) {
+
+            } else {
+                warning (err.message);
+            }
```

Comment:
Is there any reason you crush FileError.ACCES error? Same reason with the following commnet?

`// if the error is not `no access to file`, because regular user`

If so please explicit by adding a comment here too.

Reply:
Also, isn't it possible to write something like this? I commented without testing if this is grammatically correct though.

```vala
            if (!(err is FileError.ACCES)) {
                warning (err.message);
            }
```